include/aos/common/tools/memory.hpp

```diff
@@ -441,6 +441,21 @@ inline UniquePtr<T> MakeUnique(Allocator* allocator, Args&&... args)
     return UniquePtr<T>(new (allocator) T(args...), DefaultDeleter<T>(allocator));
 }
 
+/**
+ * Defers object destruction till the end of the current scope.
+ *
+ * @tparam T type of the object to be destroyed.
+ * @tparam Deleter type of the deleter.
+ * @param ptr pointer to the object to be destroyed.
+ * @param deleter functor object to be deferred.
+ * @return UniquePtr<T, Deleter>.
+ */
+template <typename T, typename Deleter>
```

Comment:
No need for this wrapper. We should use the deleted as follows:
```c++
auto drbg = aos::UniquePtr<mbedtls_ctr_drbg_context>(&ctrDrbg, mbedtls_ctr_drbg_free);
```

Reply:
I see gcc has some limitations on CTAD, particularary for std::unique_ptr. So, both parameters should be provided in the example above like:
auto drbg = aos::UniquePtr<mbedtls_ctr_drbg_context, void (*)(mbedtls_ctr_drbg_context*)>(&ctrDrbg, mbedtls_ctr_drbg_free); 
and it doesn't look very good.